src/include/OSL/rendererservices.h

```diff
@@ -189,6 +206,47 @@ class OSLEXECPUBLIC RendererServices {
                          const EncodedType* arg_types, uint32_t arg_values_size,
                          uint8_t* arg_values);
 
+    /// Builds a free function to provide a value for a given attribute.
+    /// This occurs at shader compile time, not at execution time.
+    ///
+    /// @param group
+    ///     The shader group currently requesting the attribute.
+    ///
+    /// @param object_lookup
+    ///     True if an object name was specified, even if the value is not
+    ///     known at compile time.
+    ///
+    /// @param object_name
+    ///     The object name, or nullptr if the value is not specified or it
+    ///     is not known at compile time.
+    ///
+    /// @param attribute_name
+    ///     The attribute name, or nullptr if the value is not known at
+    ///     compile time.
+    ///
+    /// @param array_lookup
```

Comment:
To match the `AttributeSpecBuiltinArg::IsArrayLookup`, can we change 
`@param array_lookup`
to
`@param is_array_lookup`

Maybe we should do the same for
`bool object_lookup`
to
`bool is_object_lookup`

Reply:
Done!